src/draw/drawLabels.ts

```diff
@@ -0,0 +1,151 @@
+import { Canvas } from 'skia-canvas';
+
+import { Image } from '../Image.js';
+import type { Point } from '../geometry/index.js';
+import { validateValues } from '../utils/validators/validators.js';
+
+type Label = number | string;
+
+interface DrawLabelsOptions {
+  /**
+   *  Size and type of font.
+   */
+  font?: string;
+  /**
+   *  Font color.
+   */
+  fontColor?: number[];
+}
+/**
+ * Draws different labels on images.
+ * @param image - Image to draw labels on.
+ * @param labels - Labels to draw.
+ * @param coordinates - Coordinates where to draw labels.
+ * @param options - DrawLabelsWithCanvasOptions.
+ * @returns Image with drawn labels.
+ */
+export function drawLabels(
+  image: Image,
+  labels: Label[],
+  coordinates: Point[],
+  options: DrawLabelsOptions = {},
+) {
+  const canvas = new Canvas(image.width, image.height);
+  const { font = '12px Helvetica', fontColor = [255, 255, 255] } = options;
+
+  validateValues(fontColor, image);
+  if (coordinates.length === 0) {
+    throw new RangeError('You must specify at least one label coordinate.');
+  }
+  if (labels.length !== coordinates.length) {
+    throw new Error('Positions and labels must be arrays of the same size.');
+  }
+
+  const normalizedColor = [
+    fontColor[0] ?? 255,
+    fontColor[1] ?? 255,
+    fontColor[2] ?? 255,
+  ];
+
+  const ctx = canvas.getContext('2d');
+  ctx.font = font;
+  ctx.fillStyle = `rgba(${normalizedColor.join(',')})`;
+
+  for (let i = 0; i < labels.length; i++) {
+    const coordinate = coordinates[i % coordinates.length];
+    ctx.fillText(
+      labels[i % labels.length] as string,
+      coordinate.column,
+      coordinate.row,
+    );
+  }
+
+  layerCanvas(
+    image.getRawImage().data as ImageDataArray,
+    ctx.getImageData(0, 0, image.width, image.height).data,
+    image.channels,
+    image.bitDepth,
+  );
+  const newImage = new Image(image.width, image.height, {
+    data: image.getRawImage().data,
+    colorModel: image.colorModel,
+  });
```

Comment:
Given that you drew on the original image's data, it's kinda wrong to return a new image with the same underlying data.

If we return a new image, the original image should be cloned before drawing.